mock/mock.go

```diff
@@ -650,10 +650,13 @@ func (m *Mock) calls() []Call {
 // Arguments holds an array of method arguments or return values.
 type Arguments []interface{}
 
+// Intentionally make mock.Anything a different type from string
+type anything string
+
 const (
 	// Anything is used in Diff and Assert when the argument being tested
 	// shouldn't be taken into consideration.
-	Anything = "mock.Anything"
+	Anything = anything("mock.Anything")
```

Comment:
I'm also considering:

```go
 // Intentionally make mock.Anything a different type from string
type anything struct{}

var (
        // Anything is used in Diff and Assert when the argument being tested
        // shouldn't be taken into consideration.
       Anything = anything{}
 )
```

I believe that would make sure this is a distinct sentinel value and it shouldn't affect existing tests.